cmd/podman/exists.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"github.com/containers/libpod/cmd/podman/libpodruntime"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"os"
+)
+
+var (
+	existFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "q, quiet",
+			Usage: "silence output including error messages",
+		},
+	}
+)
+
+var (
+	imageExistsDescription = `
+	podman image exists 
+
+	Check if an image exists in local storage
+`
+
+	imageExistsCommand = cli.Command{
+		Name:         "exists",
+		Usage:        "Check if an image exists in local storage",
+		Description:  imageExistsDescription,
+		Action:       imageExistsCmd,
+		ArgsUsage:    "IMAGE-NAME",
+		OnUsageError: usageErrorHandler,
+	}
+)
+
+var (
+	containerExistsDescription = `
+	podman container exists 
+
+	Check if a container exists in local storage
+`
+
+	containerExistsCommand = cli.Command{
+		Name:         "exists",
+		Usage:        "Check if a containerexists in local storage",
+		Description:  containerExistsDescription,
+		Action:       containerExistsCmd,
+		ArgsUsage:    "CONTAINER-NAME",
+		OnUsageError: usageErrorHandler,
+	}
+)
+
+func imageExistsCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) > 1 || len(args) < 1 {
+		return errors.New("you may only check for the existence of one image at a time")
+	}
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+	if _, err := runtime.ImageRuntime().NewFromLocal(args[0]); err != nil {
+		if !c.Bool("quiet") {
+			os.Stderr.WriteString(err.Error())
+		}
+		os.Exit(1)
+	}
+	return nil
+}
+
+func containerExistsCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) > 1 || len(args) < 1 {
+		return errors.New("you may only check for the existence of one container at a time")
+	}
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+	if _, err := runtime.LookupContainer(args[0]); err != nil {
```

Comment:
If the error isn't ErrNoSuchCtr or ErrCtrInvalid we should return an exit code that's not 1

Reply:
Correct the exit code should be set to 1 in only those cases.  Otherwise return the error and allow podman to exit with 125 like it normally does.
I don't think we should report an error when it does not exist at all.  Since the tool sort of expects it not to exist, it should not report it as such.  Now if it got an error other then container does not exist, we want to report the error.  And thus we should remove the --quiet option.

Reply:
fixed.